src/swarm/neo/connection/RequestOnConnBase.d

```diff
@@ -507,6 +614,185 @@ abstract class RequestOnConnBase
+                this.outer.send_payload_ = this.outer.send_payload;
+
+                // registerForSending may fail if called while connection
+                // shutdown is in progress
+                scope ( failure )
+                    this.outer.send_payload_ = null;
+
+                if (!this.outer.connection.registerForSending(
+                    this.outer.request_id))
+                {
+                    assert(false, "nextEvent: already sending");
+                }
+
+                // Sending may succeed immediately. send_payload_ is null, in
+                // this case.
+                if ( !this.outer.send_payload_ )
+                {
+                    fired_event.sent = Sent();
+                    return fired_event;
+                }
+            }
+
+            scope ( exit )
+            {
+                if ( sending )
+                    this.outer.connection.unregisterForSending(
+                        this.outer.request_id);
+            }
+
+            // Set up receiving, if required.
+            if ( receiving )
+            {
+                // registerForErrorNotification may fail if called while
+                // connection shutdown is in progress
+                scope ( failure )
+                    this.outer.recv_payload_ = null;
+
+                if (!this.outer.connection.registerForErrorNotification(
+                    this.outer.request_id))
+                {
+                    assert(false, "nextEvent: already receiving");
+                }
+            }
+
+            scope ( exit )
+            {
+                if ( receiving )
+                    this.outer.connection.unregisterForErrorNotification(
+                        this.outer.request_id);
+            }
+
+            // Set up yielding, if required.
+            scope yielded_rqonconn = this.outer.new YieldedRequestOnConn;
+
+            if ( yielding )
+            {
+                if (!this.outer.yielded_rqonconns.add(yielded_rqonconn))
+                    assert(false, "nextEvent: already added to the resumer");
+            }
+
+            scope ( exit )
+            {
+                if ( yielding )
+                    this.outer.yielded_rqonconns.remove(yielded_rqonconn);
+            }
+
+            // Suspend the fiber and wait for the next event to occur.
+            int resume_code = this.outer.suspendFiber();
+            switch ( resume_code )
+            {
+                case FiberResumeCode.Sent:
+                    if ( sending )
+                    {
+                        fired_event.sent = Sent();
+                        sending = false; // don't unregister on exit; already done
+                    }
+                    else
+                    {
+                        auto e = this.protocol_error.set(
+                            "RequestOnConn unexpectedly resumed with Sent");
+                        this.shutdownConnection(e);
```

Comment:
I think so. To be honest, I just copied this from the old functions, so I've not re-checked exactly what it does.